tests/script/test_evaluate.py

```diff
@@ -395,3 +397,66 @@ async def test_scriptEvaluate_realm(websocket, context_id):
         "exceptionDetails": ANY,
         "realm": realm
     } == result
+
+
+@pytest.mark.asyncio
+async def test_scriptEvaluate_dedicated_worker(websocket, context_id, html,
+                                               snapshot):
+    worker_url = 'data:application/javascript,'
+    url = html(f"<script>window.w = new Worker('{worker_url}');</script>")
+
+    await subscribe(websocket, ["script.realmCreated"])
+
+    await send_JSON_command(
+        websocket, {
+            "method": "browsingContext.navigate",
+            "params": {
+                "context": context_id,
+                "url": url,
+                "wait": "complete",
+            }
+        })
+
+    # Wait for worker to be created
+    while True:
+        message = await wait_for_event(websocket, "script.realmCreated")
+        if message["params"] == {
+                "realm": ANY_STR,
+                "origin": worker_url,
+                "type": "dedicated-worker"
+        }:
+            realm = message["params"]["realm"]
+            break
+
+    # Set up a listener on the page.
+    id = await send_JSON_command(
+        websocket, {
+            "method": "script.evaluate",
+            "params": {
+                "target": {
+                    "context": context_id
+                },
+                "expression": "new Promise(resolve => window.w.addEventListener('message', ({data}) => resolve(data), {once: true}))",
+                "awaitPromise": True
+            }
+        })
+
+    # Post a message from the worker.
+    await execute_command(
```

Comment:
This should be `send_JSON_command`, as I am not sure if the order of completion will always be correct -> You may get the resolve before the command is completed.